src/Auth/CookieTokenAuthenticate.php

```diff
@@ -91,7 +91,7 @@ public function authenticate(ServerRequest $request, Response $response)
     {
         // Only attempt to authenticate once per session
         if (!$this->authenticateAttemptedThisSession($request)) {
-            if ($this->_config['minimizeCookieExposure']) {
+            if ($this->_config['minimizeCookieExposure'] && !$request->is('ajax')) {
```

Comment:
There might be other cases where a developer wants to disable the redirect as well. Perhaps being able to configure a callback (predicate) taking the request as input and outputting a boolean would be a good long-term solution.

For now I'll take this, though. 👍

Reply:
This sounds like a good idea. I wasn't crazy about throwing that all inclusive isAjax test in there. I will look into implementing the callback...